java/junit5/src/main/java/com/saucelabs/saucebindings/junit5/SauceBindingsExtension.java

```diff
@@ -1,114 +1,171 @@
   }
 
-  public SauceBindingsExtension(SauceOptions sauceOptions) {
-    this(sauceOptions, DataCenter.US_WEST);
-  }
-
-  public SauceBindingsExtension(SauceOptions sauceOptions, DataCenter dataCenter) {
+  private SauceBindingsExtension(SauceOptions sauceOptions, DataCenter dataCenter) {
     this.sauceOptions = sauceOptions;
     this.dataCenter = dataCenter;
+    this.buildName = CITools.getBuildName() + ": " + CITools.getBuildNumber();
   }
 
-  public SauceBindingsExtension(Capabilities capabilities) {
-    this(capabilities, DataCenter.US_WEST);
-  }
-
-  public SauceBindingsExtension(Capabilities capabilities, DataCenter dataCenter) {
-    this.sauceOptions = new SauceOptions();
-
-    // TODO: Update Sauce Bindings to handle "sauce:options" the same as it handles "sauce
-    Map<String, Object> capabilitiesMap = new HashMap<>(capabilities.asMap());
-    Optional.ofNullable(capabilitiesMap.get("sauce:options"))
-        .filter(Map.class::isInstance)
-        .map(Map.class::cast)
-        .ifPresent(
-            sauceOptionsMap -> {
-              capabilitiesMap.put("sauce", sauceOptionsMap);
-              capabilitiesMap.remove("sauce:options");
-            });
-
-    this.sauceOptions.mergeCapabilities(capabilitiesMap);
-    this.dataCenter = dataCenter;
-  }
+  @Override
+  public void beforeEach(ExtensionContext context) {
+    SauceOptions options = updateOptions(context);
+    SauceSession session = new SauceSession(options);
+    session.setDataCenter(dataCenter);
+    WebDriver driver = session.start();
 
-  public SauceBindingsExtension(DataCenter dataCenter) {
-    this(new SauceOptions(), dataCenter);
+    getStore(context).put("session", session);
+    getStore(context).put("driver", driver);
   }
 
-  @Override
-  public void beforeEach(ExtensionContext context) {
-    if (isExtensionDisabled()) {
-      return;
+  private SauceOptions updateOptions(ExtensionContext context) {
+    SauceOptions options = sauceOptions.copy();
+    options.sauce().setName(context.getDisplayName());
+    options.sauce().getCustomData().put("sauce-bindings", "junit5");
+    List<String> tags = options.sauce().getTags();
+    if (tags != null) {
+      tags.addAll(context.getTags());
+    } else {
+      options.sauce().setTags(new ArrayList<>(context.getTags()));
     }
+    options.sauce().setBuild(buildName);
 
-    if (sauceOptions.sauce().getName() == null) {
-      sauceOptions.sauce().setName(context.getDisplayName());
-    }
+    return options;
+  }
 
-    session = new SauceSession(sauceOptions);
-    session.setDataCenter(dataCenter);
-    driver = session.start();
+  private ExtensionContext.Store getStore(ExtensionContext context) {
+    return context.getStore(
+        ExtensionContext.Namespace.create(getClass(), context.getRequiredTestMethod()));
   }
 
   @Override
   public void testSuccessful(ExtensionContext context) {
-    if (isExtensionDisabled()) {
-      return;
+    if (!SauceSession.isDisabled()) {
```

Comment:
1. Implement it in one place instead of each of the libraries
2. It's much easier to deal with empty and null returns when disable code is in Bindings